nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -241,109 +170,74 @@ nav2_util::CallbackReturn
 BtNavigator::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
 {
   RCLCPP_INFO(get_logger(), "Shutting down");
+
+  if (!bt_action_server_->on_shutdown()) {
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
-void
-BtNavigator::navigateToPose()
+bool
+BtNavigator::onGoalReceived(Action::Goal::ConstSharedPtr goal)
 {
-  initializeGoalPose();
-
-  auto is_canceling = [this]() {
-      if (action_server_ == nullptr) {
-        RCLCPP_DEBUG(get_logger(), "Action server unavailable. Canceling.");
-        return true;
-      }
-
-      if (!action_server_->is_server_active()) {
-        RCLCPP_DEBUG(get_logger(), "Action server is inactive. Canceling.");
-        return true;
-      }
-
-      return action_server_->is_cancel_requested();
-    };
+  RCLCPP_INFO(
+    get_logger(), "Begin navigating from current location to (%.2f, %.2f)",
+    goal->pose.pose.position.x, goal->pose.pose.position.y);
 
-  std::string bt_xml_filename = action_server_->get_current_goal()->behavior_tree;
+  // Reset state for new action feedback
+  start_time_ = now();
 
-  // Empty id in request is default for backward compatibility
-  bt_xml_filename = bt_xml_filename.empty() ? default_bt_xml_filename_ : bt_xml_filename;
+  auto bt_xml_filename = goal->behavior_tree;
 
-  if (!loadBehaviorTree(bt_xml_filename)) {
+  if (!bt_action_server_->loadBehaviorTree(bt_xml_filename)) {
```

Comment:
I see this is moved into the `onGoalReceived` from the main function. I don't see where the default vs current check is made https://github.com/ros-planning/navigation2/blob/main/nav2_bt_navigator/src/bt_navigator.cpp#L269-L277

Reply:
That's been moved to the `loadBehaviorTree` function itself:
https://github.com/naiveHobo/navigation2/blob/3a600c5f6be44a378962ffabde926eda56b63bdb/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp#L149

If this function receives an empty filename, it will load the default file.